torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -54,6 +54,11 @@ def init(self):
         # Reduce-scatter stream gives separate execution "thread" for post-
         # backward logic like pre/post-gradient division and reduce-scatter
         self.reduce_scatter_stream = torch.cuda.Stream(priority=high_priority)
+        # run the HSDP all-reduces concurrently with all-gather/reduce-scatter
+        # since collectives use different network resources and can overlap
+        # in the typical intra-node sharding / inter-node replication case
+        # TODO: understand if HSDP needs stream(high_priority)
```

Comment:
I think it is not needed. We actually prefer all-reduce to be lower priority since it does not block anything.